core/kernel.rbs

```diff
@@ -1564,7 +1572,14 @@ module Kernel : BasicObject
   #     (snipped)
   #     ping
   #
-  def self?.select: (::Array[IO] read, ?::Array[IO] write, ?::Array[IO] error, ?Time::_Timeout timeout) -> ::Array[String]
+  def self?.select: ([]?, ?[]?, ?[]?, ?nil) -> bot
+                  | [R < _ToIO]                       (Array[R] read, ?[]?,            ?[]?,            ?nil                  ) -> [Array[R], [], []]
+                  | [R < _ToIO]                       (Array[R] read,  []?,             []?,            Time::_Timeout timeout) -> [Array[R], [], []]?
+                  | [R < _ToIO, W < _ToIO]            (Array[R] read, ?Array[W] write, ?[]?,            ?nil                  ) -> [Array[R], Array[W], []]
+                  | [R < _ToIO, W < _ToIO]            (Array[R] read,  Array[W] write,  []?,            Time::_Timeout timeout) -> [Array[R], Array[W], []]?
+                  | [R < _ToIO, W < _ToIO, E < _ToIO] (Array[R] read, ?Array[W] write, ?Array[E] error, ?nil                  ) -> [Array[R], Array[W], Array[E]]
+                  | [R < _ToIO, W < _ToIO, E < _ToIO] (Array[R] read,  Array[W] write,  Array[E] error, Time::_Timeout timeout) -> [Array[R], Array[W], Array[E]]?
+                  | (Array[_ToIO]? read, ?Array[_ToIO]? write, ?Array[_ToIO]? error, ?Time_Timeout? timeout) -> [Array[_ToIO], Array[_ToIO], Array[_ToIO]]?
```

Comment:
The `?[]?` is because you can have empty arrays or `nil` as options for those arguments; the timeout can only be`nil`, or unspecified though.